lib/queuePopulator/LogReader.js

```diff
@@ -291,6 +293,15 @@ class LogReader {
             timeoutMs: params.timeoutMs,
             logger: this.log.newRequestLogger(),
         };
+        // When using the RaftLogReader log consumer, The batch
+        // processing can get stuck sometimes for unknown reasons.
+        // As a temporary fix we set a timeout to kill the process.
+        const batchTimeoutTimer = setTimeout(() => {
+            this.log.error('queue populator batch timeout', {
+                logStats: batchState.logStats,
+            });
+            process.emit('SIGTERM');
```

Comment:
should we really "crash" ?
would it not be better to fail healthcheck? (may not be supported at the moment in s3c, but will need to anyway...)